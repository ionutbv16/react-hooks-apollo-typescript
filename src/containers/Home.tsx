import React from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import styled from 'styled-components';
import CardComp from '../components/Card';
import {PARAM_PEOPLE, PARAM_STARSHIPS} from '../config';

const Wrapper = styled.div`
  width: 100%;
  overflow-y: auto;
  padding: 20px;
`;

const ActionBar = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
  padding: 14px;
`;

const Home: React.FC<RouteComponentProps> = () => {

  return (
    <Wrapper>
      <ActionBar>
      Start the game by selecting one of the options bellow:
      </ActionBar>
      <ActionBar>
        <CardComp name={PARAM_PEOPLE} /> 
        <CardComp name={PARAM_STARSHIPS} /> 
      </ActionBar>
    </Wrapper>
  );
};

export default withRouter(Home);
