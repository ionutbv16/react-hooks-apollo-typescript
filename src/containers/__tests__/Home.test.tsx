import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home'; 
import { MemoryRouter } from 'react-router';
 
describe('Test Home Page', () => {
  it('should render Home without error', () => {
    const component = renderer.create(<MemoryRouter><Home /></MemoryRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
