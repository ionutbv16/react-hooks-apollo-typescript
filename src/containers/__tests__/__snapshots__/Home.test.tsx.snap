// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page should render Home without error 1`] = `
<div
  className="sc-EHOje bfoJug"
>
  <div
    className="sc-bZQynM khFbRa"
  >
    Start the game by selecting one of the options bellow:
  </div>
  <div
    className="sc-bZQynM khFbRa"
  >
    <div
      className="sc-bdVaJa kgqLdc"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <div
          className="sc-bwzfXH fdVHAP"
        >
          <a
            href="/game/People"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX bZbatV"
            >
              People
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa kgqLdc"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <div
          className="sc-bwzfXH fdVHAP"
        >
          <a
            href="/game/Starships"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX bZbatV"
            >
              Starships
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render Home without error 1`] = `
<div
  className="sc-EHOje bfoJug"
>
  <div
    className="sc-bZQynM khFbRa"
  >
    Start the game by selecting one of the options bellow:
  </div>
  <div
    className="sc-bZQynM khFbRa"
  >
    <div
      className="sc-bdVaJa kgqLdc"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <div
          className="sc-bwzfXH fdVHAP"
        >
          <a
            href="/game/People"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX bZbatV"
            >
              People
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa kgqLdc"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <div
          className="sc-bwzfXH fdVHAP"
        >
          <a
            href="/game/Starships"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX bZbatV"
            >
              Starships
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
